test/unit/query/builder.js

```diff
@@ -51,12 +64,11 @@ function verifySqlResult(dialect, expectedObj, sqlObj) {
   });
 }
 
-function testsql(chain, valuesToCheck, selectedClients) {
+function testsql(queryBuilderCallback, valuesToCheck, selectedClients) {
   selectedClients = selectedClients || clients;
   Object.keys(valuesToCheck).forEach(function(key) {
-    var newChain = chain.clone()
-        newChain.client = selectedClients[key]
```

Comment:
Is there some reason why setting the client didn't work? if setting client in chain doesn't affect all subqueries etc, then it is another bug in knex.